### PR TITLE
Server recordings 2: Integrated with `Admin::ServerRecordings#index` API.

### PR DESCRIPTION
**Pull request by KH-Amir-TN**
## Description

Enable users with the right set of privileges to view server recordings.

This PR completes **2.** 
--
~~0. Add `Admin::ServerRecordings#index`.~~
~~1. Extend Recordings UIs.~~
~~2. Add ServerRecordings UI. & Integrate the UI with the API.~~
 
User Story [View Server Recordings]:
---
0. User with the right set of permissions.
1. User navigates to Server Recordings.
2. User can view, filter and sort server recordings.

## Testing Steps

> 1. Pull the code.
> 2. Install the dependencies `bundle install && npm|yarn install`.
> 3. Clean the previous assets build by running `rm app/assets/builds/*` (This won't remove .keep since it's hidden).
> 4. Clean the database and tmp files for a better isolation by running `rails tmp:clear && rails db:schema:cache:clear && rails db:drop && rails db:create && rails db:migrate:with_data`
> 5. Run the linter and specs `bundle exec rubocop --parallel && bundle exec rspec && npx eslint app/javascript/* --ext .jsx,.js`
> 6. Run `./bin/dev` to run the assets builders processes and the Puma server all at once.
## Screenshots (if appropriate):

image
image
image

### CODE REVIEW

**Comment by KH-Amir-TN** [comment]
# DEPENDS ON #3637

File: app/javascript/components/admin/server-recordings/ServerRecordingRow.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import useUpdateRecording from '../../../hooks/mutations/recordings/useUpdateRecording';
+import useDeleteRecording from '../../../hooks/mutations/recordings/useDeleteRecording';
+
+export default function ServerRecordingRow({ recording }) {
+  // TODO: Change all mutations when their APIs becomes ready.
+  return (
+    <RecordingRow
+      recording={recording}
+      visibilityMutation={useUpdateRecordingVisibility}
+      updateMutation={useUpdateRecording}
```
Comment (KH-Amir-TN) [inline app/javascript/components/admin/server-recordings/ServerRecordingRow.jsx:17]: @hadicheaito1 for #3258 inject your mutation in lace of `useUpdateRecording`.

Comment (hadicheaito1) [reply app/javascript/components/admin/server-recordings/ServerRecordingRow.jsx:14]: Perfect, thanks Amir!

**Review by hadicheaito1 — APPROVED** [review]

File: app/javascript/components/admin/server-recordings/ServerRecordingRow.jsx (hunk 1 above)
Comment (KH-Amir-TN) [reply app/javascript/components/admin/server-recordings/ServerRecordingRow.jsx:14]: You're welcome ^^

**Pull request merged**